apps/mobile/src/features/send/forms/stx/sip10-loader.tsx

```diff
@@ -0,0 +1,94 @@
+import { Error } from '@/components/error/error';
+import { type FetchState, toFetchState } from '@/components/loading/fetch-state';
+import { SendFormLoadingSpinner } from '@/features/send/components/send-form-layout';
+import {
+  useSip10AccountBalance,
+  useSip10AddressBalanceQuery,
+} from '@/queries/balance/sip10-balance.query';
+import '@/queries/balance/stx-balance.query';
+import { useMarketDataQuery } from '@/queries/market-data/market-data.query';
+import { useNextNonce } from '@/queries/stacks/nonce/account-nonces.hooks';
+import { useStacksSignerAddressFromAccountIndex } from '@/store/keychains/stacks/stacks-keychains.read';
+import { useQueryClient } from '@tanstack/react-query';
+
+import { AccountId, FungibleCryptoAsset, MarketData, Money } from '@leather.io/models';
+import { Sip10Balance } from '@leather.io/services';
+
+interface Sip10Data {
+  availableBalance: Money;
+  quoteBalance: Money;
+  nonce: number | undefined;
+  marketData: MarketData;
+}
+
+function useSip10Data({
+  fingerprint,
+  accountIndex,
+  token,
+}: { token: Sip10Balance } & AccountId): FetchState<Sip10Data> {
+  const address = useStacksSignerAddressFromAccountIndex(fingerprint, accountIndex) ?? '';
+  const balance = useSip10AddressBalanceQuery(address);
+  const nextNonce = useNextNonce(address);
+  const marketData = useMarketDataQuery(token.asset);
+
+  // TODO: Replace with aggregate queries once we have more flexible query API
+  const isReady =
+    balance.status === 'success' &&
+    marketData.status === 'success' &&
+    nextNonce.status === 'success';
+  const isLoading =
+    balance.status === 'pending' ||
+    marketData.status === 'pending' ||
+    nextNonce.status === 'pending';
+  const isError =
```

Comment:
It would be great to add tracking of these errors, particularly the `nonce` error as the `Granite` team were complaining about that to me. 

They also complain that often transactions fail silently and I think it can be doe to not getting the `nextNonce`